dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04205MigrateVanityURLToContent.java

```diff
@@ -0,0 +1,251 @@
+ * virtual_link fields are migrated like this: Title -> Title Url -> This is splited, one is the
+ * Site and the other one the Uri Uri -> Forward To
+ *
+ * @author erickgonzalez
+ * @version 4.2.0
+ * @since Jun 8, 2017
+ */
+public class Task04205MigrateVanityURLToContent extends AbstractJDBCStartupTask {
+
+    private static final String SELECT_VIRTUAL_LINK_QUERY = "select inode, title, url, uri from virtual_link";
+    private static final String UPDATE_INODE_QUERY = "update inode set type = 'contentlet' where type = 'virtual_link' ";
+    private static final String SELECT_HOSTID_BY_NAME_QUERY = "select identifier from contentlet where text1 = ?";
+    private static final String SELECT_HOSTLANG_BY_ID_QUERY = "select language_id from contentlet where identifier = ?";
+    private static final String INSERT_IDENTIFIER_QUERY = "insert into identifier(id, parent_path, "
+    +"asset_name, host_inode, asset_type) values (?,?,?,?,?)";
+    private String INSERT_CONTENTLET_QUERY =
+            "insert into contentlet(inode, show_on_menu, sort_order, title, mod_date, mod_user, "
+            +"friendly_name, structure_inode, identifier, language_id, text1, text2, text3, text4"
+            +", integer1, integer2, integer3, integer4, integer15, integer16, integer17, integer18"
+            +",integer19, integer20, integer21, integer22, integer23, integer24, integer25,"
+            +"\"float1\", \"float2\", \"float3\", \"float4\", \"float5\", \"float6\", \"float7\","
+            +"\"float8\", \"float9\", \"float10\", \"float11\", \"float12\","
+            +"\"float13\", \"float14\", \"float15\", \"float16\", \"float17\", \"float18\","
+            +"\"float19\", \"float20\", \"float21\", \"float22\", \"float23\", \"float24\",\";"
+            +"\"float25\", bool1, bool2, bool3, bool4, bool5, bool6, bool7, bool8, bool9, bool10,"
+            +" bool11, bool12, bool13, bool14, bool15, bool16, bool17, bool18, bool19, bool20,"
+            +" bool21, bool22, bool23, bool24, bool25) "
+            +"values (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,"
+            +"?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,"
+            +"?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)";
+    private static final String INSERT_CONTENTLET_VERSION_INFO_QUERY =
+            "insert into contentlet_version_info(identifier, lang, working_inode, live_inode, deleted, locked_on, version_ts)"
+                    + "values (?,?,?,?,?,?,?)";
+    private static final String DELETE_INODE_BY_INODE_QUERY = "delete from inode where inode = ?";
+    private static final String DELETE_VIRTUAL_LINK_QUERY = "delete from virtual_link";
+
+    private static final int CODE_301 = 301;
+    private static final int FOR_LIMIT_48 = 48;
+    private static final int FOR_LIMIT_25 = 25;
+    private static final int INT_2 = 2;
+
+    @Override
+    public void executeUpgrade() throws DotDataException, DotRuntimeException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)